src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -433,6 +427,10 @@ private async Task<WorkUnitResult> ExecuteBucket(TaskHost taskHost, ItemBucket b
                     if (requirements != null)
                     {
                         TaskLoggingContext taskLoggingContext = _targetLoggingContext.LogTaskBatchStarted(_projectFullPath, _targetChildInstance);
+                        if (MSBuildEventSource.Log.IsEnabled())
```

Comment:
nit: The `IsEnabled` check is likely not needed here. Evaluating the arguments is just a few dereferences. I would either remove it or cache the result in a local so it doesn't have to be called twice.

Reply:
Agreed; this was a more complex call before when I first added it.

That said, I don't think it matters. My understanding is that the check is so fast that should basically consider it a no-op as far as time taken, and ExecuteTaskStart implicitly calls IsEnabled after it's evaluated its arguments, so they should be equivalent from a correctness perspective. I'll still remove it. Thanks!